### PR TITLE
[USM] Add raw tests for handling RST/EOS

### PR DESCRIPTION
**Pull request by amitslavin**
### What does this PR do?

Add raw tests to validate our EOS/RST support.

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided. Except if the `qa/skip-qa` label, with required either `qa/done` or `qa/no-code-change` labels, are applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature.
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: pkg/network/usm/usm_http2_monitor_test.go [hunk 1]
```diff
@@ -227,6 +227,69 @@ func (s *usmHTTP2Suite) TestRawTraffic() {
 				}: 5,
 			},
 		},
+		{
+			name: "validate RST_STREAM cancel err code",
+			// The purpose of this test is to validate that when a cancel error code is sent, we will not count the request.
+			messageBuilder: func() []byte {
+				const headerFramesCount = 10
+				const rstCancelFramesCount = 5
+				const rstCancelFramesCountInMessage = 1
```
Comment (guyarb) [inline pkg/network/usm/usm_http2_monitor_test.go:236]: Not sure the naming is good enought

File: pkg/network/usm/usm_http2_monitor_test.go [hunk 2]
```diff
@@ -521,3 +584,55 @@ func createMessageWithPingAndWindowUpdate(t *testing.T, headerFields []hpack.Hea
...
+			StreamID:      uint32(streamID),
+			BlockFragment: headersFrame,
+			EndStream:     false,
+			EndHeaders:    true,
+		}), "could not write header frames")
+
+		if len(includeStatusCode) > 0 && includeStatusCode[0] {
+			statusOkHeaderField := []hpack.HeaderField{
+				{Name: ":status", Value: "200"},
+			}
```
Comment (guyarb) [inline pkg/network/usm/usm_http2_monitor_test.go:611]: that's a server response, how can you write it as a client?

File: pkg/network/usm/usm_http2_monitor_test.go [hunk 3]
```diff
@@ -521,3 +584,55 @@ func createMessageWithPingAndWindowUpdate(t *testing.T, headerFields []hpack.Hea
...
+				EndStream:     false,
+				EndHeaders:    true,
+			}), "could not write header frames")
+		}
+
+		// Writing the data frame to the buffer using the Framer.
+		require.NoError(t, framer.WriteData(uint32(streamID), true, []byte{}), "could not write data frame")
+
+		if rstFramesCount > 0 {
+			for j := 0; j < rstFramesCountInMessage; j++ {
```
Comment (guyarb) [inline pkg/network/usm/usm_http2_monitor_test.go:629]: why is the outer condition is `rseFramesCount > 0` but the for loop uses `rstFramesCountInMessage`?

**Review by guyarb — CHANGES_REQUESTED** [review]

File: pkg/network/usm/usm_http2_monitor_test.go (hunk 2 above)
Comment (amitslavin) [reply pkg/network/usm/usm_http2_monitor_test.go]: removed

File: pkg/network/usm/usm_http2_monitor_test.go (hunk 3 above)
Comment (amitslavin) [reply pkg/network/usm/usm_http2_monitor_test.go]: what do you think now?

File: pkg/network/usm/usm_http2_monitor_test.go (hunk 1 above)
Comment (amitslavin) [reply pkg/network/usm/usm_http2_monitor_test.go]: what about now?

**Review by guyarb — APPROVED** [review]

**Pull request merged**